pkg/opendsme/contrib/DSMEMessage.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2022 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  José I. Álamos <jose.alamos@haw-hamburg.de>
+ */
+#include "assert.h"
+#include "opendsme/DSMEMessage.h"
+#include "net/gnrc/pktdump.h"
+#include "net/gnrc.h"
+#include "opendsme/opendsme.h"
+
+namespace dsme {
+
+void DSMEMessage::prependFrom(DSMEMessageElement *msg)
+{
+    gnrc_pktsnip_t *head = gnrc_pktbuf_add(pkt, NULL,
+                                           msg->getSerializationLength(), GNRC_NETTYPE_UNDEF);
+
+    DSME_ASSERT(msg->getSerializationLength() <= IEEE802154_FRAME_LEN_MAX);
```

Comment:
I'm not aware of any openDSME limitations for that, so I guess it should be possible :)

Reply:
since the implementation assumes O-QPSK, I would propose to keep it.
At some point one of the functions returned a corrupted packet, which made `msg->getSerializationLength` return a value higher than 127.
This assertion makes sure that nothing went wrong